tests/hw/esp-tester/tester.toit

```diff
@@ -124,19 +124,31 @@ run-test invocation/cli.Invocation:
     board1-ready := monitor.Latch
 
     task::
-      log "Connecting to board1"
-      board1.connect-network
+      image/ByteArray? := null
+      Task.group [
+        ::
+          log "Connecting to board1"
+          board1.connect-network,
+        ::
+          image = board1.compile-test test-path,
+      ]
       log "Installing test on board1"
-      board1.install-test test-path arg
+      board1.install-test image arg
       log "Board1 ready"
       board1-ready.set true
 
     if port-board2:
       board2 = TestDevice --name="board2" --port-path=port-board2 --ui=ui --toit-exe=toit-exe
-      log "Connecting to board2"
-      board2.connect-network
+      image2/ByteArray? := null
+      Task.group [
+        ::
+          log "Connecting to board2"
```

Comment:
Maybe move the logging to `connect-network`? The `TestDevice` already knows its name and the `Task.group` call would look even better.